src/store/holochain.js

```diff
@@ -223,10 +249,27 @@ export default {
           }
         }
       }, 1000)
+
+      // timeout initalize client attempt loop at 30min
+      const initalizeClientTimeout = 1_800_000 // 30min in ms
+      const timeoutId = setTimeout(() => {
+        if (!(state.holochainClient || state.holoClient)){
+          console.error(`Could not initialize ${isHoloHosted() ? 'holo' : 'holochain'} client. Timed out at ${initalizeClientTimeout} ms`)
+          clearInterval(intervalId)
+          commit('resetConnectionState')
+          commit('setReconnecting', 0)
+        } else {
+          clearTimeout(timeoutId)
+        }
```

Comment:
I don't like having an if/else with negated condition. 
```suggestion
        if (state.holochainClient || state.holoClient) {
          clearTimeout(timeoutId)
        } else {
          console.error(`Could not initialize ${isHoloHosted() ? 'holo' : 'holochain'} client. Timed out at ${initalizeClientTimeout} ms`)
          clearInterval(intervalId)
          commit('resetConnectionState')
          commit('setReconnecting', 0)
        }
```